tests/test_solarposition.py

```diff
@@ -0,0 +1,343 @@
+        [32.21696737, 32.24355729, 57.78303263, 57.75644271, 204.91857639],
+    ]
+    data = pd.DataFrame(data=values, columns=columns)
+    return data
+
+
+def expected_walraven():
+    columns = ['elevation', 'zenith', 'azimuth']
+    values = [
+        [32.21577184, 57.78422816, 204.9145408],
+        [32.20330752, 57.79669248, 204.96082497],
+        [34.91988403, 55.08011597, 169.37445826],
+        [18.63514306, 71.36485694, 234.18579686],
+        [35.75942569, 54.24057431, 197.67543014],
+        [-9.53241956, 99.53241956, 201.18624892],
+        [66.85832643, 23.14167357, 245.07813388],
+        [9.53241956, 80.46758044, 338.81375108],
+        [50.11302395, 39.88697605, 326.23525905],
+        [35.35901685, 54.64098315, 175.38665251],
+        [-53.24443195, 143.24443195, 18.64588668],
+        [-53.24443195, 143.24443195, 18.64588668],
+        [33.19945919, 56.80054081, 205.08690939],
+        [32.078221, 57.921779, 204.90430273],
+        [-23.41074972, 113.41074972, 79.55008786],
+        [1.10528968, 88.89471032, 104.541125],
+        [32.21577184, 57.78422816, 204.9145408],
+        [32.21577184, 57.78422816, 204.9145408],
+        [32.21577184, 57.78422816, 204.9145408],
+    ]
+    data = pd.DataFrame(data=values, columns=columns)
+    return data
+
+
+@pytest.fixture()
+def test_conditions():
+    inputs = pd.DataFrame(
+        columns=['time', 'latitude', 'longitude', 'altitude'],
+        data=(
+            # Units: time, degrees, degrees, m
+            ['2020-10-17T12:30+00:00', 45, 10, None],  # reference
+            ['2020-10-17T12:30:10+00:00', 45, 10, None],  # non-zero seconds
+            ['2020-10-17T12:30+02:00', 45, 10, None],  # positive timezone
+            ['2020-10-17T12:30-02:00', 45, 10, None],  # negative timezone
+            ['2020-02-29T12:30+00:00', 45, 10, None],  # leap day
+            ['2020-10-17T12:30+00:00', 90, 10, None],  # 90 degree latitude
+            ['2020-10-17T12:30+00:00', 0, 10, None],  # zero latitude
+            ['2020-10-17T12:30+00:00', -90, 10, None],  # -90 degree latitude
+            ['2020-10-17T12:30+00:00', -45, 10, None],  # negative mid-latitude
+            ['2020-10-17T12:30+00:00', 45, -15, None],  # negative longitude
+            ['2020-10-17T12:30+00:00', 45, -180, None],  # 180-degree longitude
+            ['2020-10-17T12:30+00:00', 45, 180, None],  # 180-degree longitude
+            ['1800-10-17T12:30+00:00', 45, 10, None],  # 200 years in the past
+            ['2200-10-17T12:30+00:00', 45, 10, None],  # 200 years ahead
+            ['2020-10-17T03:30+00:00', 45, 10, None],  # negative sun elevation
+            ['2020-10-17T05:50+00:00', 45, 10, None],  # ~1-deg sun elevation
+            ['2020-10-17T12:30+00:00', 45, 10, 0],  # zero altitude
+            ['2020-10-17T12:30+00:00', 45, 10, -100],  # negative altitude
+            ['2020-10-17T12:30+00:00', 45, 10, 4000],  # positive altitude
```

Comment:
@AdamRJensen maybe I miss it, but I do not see `altitude` getting used anywhere in the tests